Runtime/BlockadeLabsSkyboxGenerator.cs

```diff
@@ -591,95 +591,184 @@ internal async Task DownloadResultAsync(GetImagineResult result, bool setSkybox
 
-            AssetDatabase.Refresh();
+                if (hasDepthMap)
+                {
+                    tasks.Add(ApiRequests.DownloadFileAsync(depthMapUrl, depthTexturePath));
+                }
 
-            var colorImporter = (TextureImporter)AssetImporter.GetAtPath(texturePath);
-            colorImporter.maxTextureSize = 8192;
-            colorImporter.textureCompression = TextureImporterCompression.Uncompressed;
-            colorImporter.mipmapEnabled = false;
-            colorImporter.textureShape = TextureImporterShape.TextureCube;
-            colorImporter.SaveAndReimport();
+                await Task.WhenAll(tasks).ConfigureAwait(true);
 
-            if (hasDepthMap)
-            {
-                var depthImporter = (TextureImporter)AssetImporter.GetAtPath(depthTexturePath);
-                depthImporter.maxTextureSize = 2048;
-                depthImporter.textureCompression = TextureImporterCompression.Uncompressed;
-                depthImporter.mipmapEnabled = false;
-                depthImporter.wrapModeU = TextureWrapMode.Repeat;
-                depthImporter.wrapModeV = TextureWrapMode.Clamp;
-                depthImporter.SaveAndReimport();
+                if (_isCancelled)
+                {
+                    Directory.Delete(folderPath, true);
+                    return;
+                }
+
+                AssetDatabase.Refresh();
+
+                var colorImporter = (TextureImporter)AssetImporter.GetAtPath(texturePath);
+                colorImporter.maxTextureSize = 8192;
+                colorImporter.textureCompression = TextureImporterCompression.Uncompressed;
+                colorImporter.mipmapEnabled = false;
+                colorImporter.textureShape = TextureImporterShape.TextureCube;
+                colorImporter.SaveAndReimport();
+
+                if (hasDepthMap)
+                {
+                    var depthImporter = (TextureImporter)AssetImporter.GetAtPath(depthTexturePath);
+                    depthImporter.maxTextureSize = 2048;
+                    depthImporter.textureCompression = TextureImporterCompression.Uncompressed;
+                    depthImporter.mipmapEnabled = false;
+                    depthImporter.wrapModeU = TextureWrapMode.Repeat;
+                    depthImporter.wrapModeV = TextureWrapMode.Clamp;
+                    depthImporter.SaveAndReimport();
+                }
             }
 
-            var colorTexture = AssetDatabase.LoadAssetAtPath<Cubemap>(texturePath);
-            var depthTexture = AssetDatabase.LoadAssetAtPath<Texture>(depthTexturePath);
-            var depthMaterial = CreateDepthMaterial(colorTexture, depthTexture, setSkybox);
+            var colorTexture = AssetDatabase.LoadAssetAtPath<Cubemap>(texturePath.ToProjectPath());
+            var depthTexture = AssetDatabase.LoadAssetAtPath<Texture>(depthTexturePath.ToProjectPath());
 
-            if (depthMaterial != null)
+            if (!File.Exists(depthMaterialPath))
```

Comment:
before checking for file existence, need to check skyboxAI.DepthMaterial == null